src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -0,0 +1,583 @@
+      (org, name, _) = coord.split(':')
+      org_name_to_org_name_rev['{}:{}'.format(org, name)] = coord
+
+    for vt in invalidation_check.all_vts:
+      t = vt.target
+      if isinstance(t, JarLibrary):
+        def get_transitive_resolved_jars(my_simple_coord, resolved_jars):
+          transitive_jar_path_for_coord = []
+          if my_simple_coord in flattened_resolution:
+            for c in [my_simple_coord] + flattened_resolution[my_simple_coord]:
+              transitive_jar_path_for_coord.extend(resolved_jars[c])
+
+          return transitive_jar_path_for_coord
+
+        for jar in t.jar_dependencies:
+          simple_coord_candidate = jar.coordinate.simple_coord
+          final_simple_coord = None
+          if simple_coord_candidate in coord_to_resolved_jars:
+            final_simple_coord = simple_coord_candidate
+          elif simple_coord_candidate in result['conflict_resolution']:
+            final_simple_coord = result['conflict_resolution'][simple_coord_candidate]
+          # If still not found, look for org:name match.
+          else:
+            org_name = '{}:{}'.format(jar.org, jar.name)
+            if org_name in org_name_to_org_name_rev:
+              final_simple_coord = org_name_to_org_name_rev[org_name]
+
+          if final_simple_coord:
+            transitive_resolved_jars = get_transitive_resolved_jars(final_simple_coord, coord_to_resolved_jars)
+            if transitive_resolved_jars:
+              compile_classpath.add_jars_for_targets([t], 'default' or classifier, transitive_resolved_jars)
+
+  def _populate_results_dir(self, vts_results_dir, results):
+    # TODO(wisechengyi): currently the path contains abs path, so need to remove that before
+    # cache can be shared across machines.
+
+    with open(os.path.join(vts_results_dir, self.RESULT_FILENAME), 'w') as f:
+      json.dump(results, f)
+
+  def _load_from_results_dir(self, compile_classpath, vts_results_dir,
+                             coursier_cache_path, invalidation_check, pants_jar_path_base):
+    """
+
+    :return: True if success; False if any of the classpath is not valid anymore.
+    """
+    result_file_path = os.path.join(vts_results_dir, self.RESULT_FILENAME)
+    if not os.path.exists(result_file_path):
+      return
+
+    with open(result_file_path, 'r') as f:
+      results = json.load(f)
+      for classifier, result in results.items():
+        try:
+          self._load_json_result(classifier, compile_classpath,
+                                 coursier_cache_path,
+                                 invalidation_check,
+                                 pants_jar_path_base, result)
+        except CouriserCacheNotFound:
+          return False
+
+    return True
+
+  @classmethod
+  def _extract_dependencies_by_root(cls, result):
+    """
+    Only extracts the transitive dependencies for the given coursier resolve.
+    Note the "dependencies" field is already transitive.
```

Comment:
Is this still the case? I thought you were fixing that.

Reply:
The print size is already greatly reduced to less than 200kB  for large targets by using the coordinates rather than nested structure. See https://github.com/coursier/coursier/pull/692#issue-273073515

If not exported in json, pants would have to put it together as well.

Reply:
I don't see why it needs to be transitive. And in fact, transitivity loses information that could allow for creating a precise graph from the coursier output.

Reply:
It does not have to be. although the status quo is that ivy code in Pants will just flatten it anyway as well. 

https://github.com/wisechengyi/pants/blob/eb11ac5ad2dc8d949dad025671cf07e5e482157e/src/python/pants/backend/jvm/ivy_utils.py#L672-L678

It is true that non-transitive would preserve more info. That's why `--tree` is also added when invoke coursier similar to the ivy report.

Do you see this as a blocker, or can I iterate on this later? There are also some profiling I want to do with coursier as well.

Reply:
No, it's not a blocker... you have my shipit. But would be good to fix in future, as we have been hoping to get a complete multigraph out of ivy/coursier, which would allow us to do one resolve and get enough information to compile with multiple versions.